src/modals/TransactionDetailsModal.tsx

```diff
@@ -152,20 +152,29 @@ const TransactionDetailsModal = ({ transaction, address, onClose }: TransactionD
           </DetailsRow>
           <DetailsRow label={t`Inputs`}>
             <AddressList>
-              {transaction.inputs?.map((input) => (
-                <ActionLinkStyled key={`${input.outputRef.key}`} onClick={() => handleShowAddress(input.address)}>
-                  <AddressEllipsed key={`${input.outputRef.key}`} addressHash={input.address} />
-                </ActionLinkStyled>
-              ))}
+              {transaction.inputs?.map(
+                (input) =>
+                  input.address !== undefined && (
+                    <ActionLinkStyled
+                      key={`${input.outputRef.key}`}
+                      onClick={() => handleShowAddress(input.address ?? '')}
+                    >
+                      <AddressEllipsed key={`${input.outputRef.key}`} addressHash={input.address} />
+                    </ActionLinkStyled>
+                  )
+              )}
```

Comment:
`input.address` is of type `string | undefined`. We don't specifically have to check for `!== undefined`. We also don't need to pass an empty string to a function that expects an addressHash (`?? ''`). The compiler does not complain with what I proposed earlier:


```ts
{transaction.inputs?.map(
                (input) =>
                  input.address && (
                    <ActionLinkStyled
                      key={`${input.outputRef.key}`}
                      onClick={() => handleShowAddress(input.address as string)}
                    >
                      <AddressEllipsed key={`${input.outputRef.key}`} addressHash={input.address} />
                    </ActionLinkStyled>
                  )
              )}
```